internal/collections/named.go

```diff
@@ -101,11 +101,41 @@ type NamedCollectionNames struct {
 }
 
 func (c *NamedCollectionNames) FindRegex(key *regexp.Regexp) []types.MatchData {
-	panic("selection operator not supported")
+	var res []types.MatchData
```

Comment:
Is there any chance `data` is empty? if so I would handle the empty case before this allocation.

Reply:
Yes, there are probably situations where `data` can be empty (I haven't tested, but I'd expect a collection like `XML` to have an empty `data` on a non-XML request )

AFAIK, declaring a slice like this does not perform any actual allocation (other than the header of the slice, which will be all zero), and the actual allocation will be performed the first time we append to it, but I can add a check on `data` if you are worried about it.